sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -288,6 +288,11 @@ public synchronized void close() {
           source.stream.getName(),
           fraction);
 
+      if (fraction <= 0.0 || fraction >= 1.0) {
+        LOG.info("BigQuery Storage API does not support splitting at fraction {}", fraction);
```

Comment:
What values did you observe ? It's surprising if you receive negative values here for example.

Reply:
The observed value was 1.0.